### ISSUE TITLE
oC10 Server Admin Manual: Recommended NFS mount options

### ISSUE THREAD

**Issue by dragotin** [owncloud/docs#3861]
## WHAT Needs to be Documented?

The docs should have a recommendation of mount-options for NFS mounted file systems.

In particular, the setting `async` vs `sync` should be mentioned as caveat as it has a big impact on performance. 

## WHY Should This Change Be Made?

Give admins a clear recommendation how to set up the server.

@orgaralf might be able to give more insights on the various mount options and how they are important for the ownCloud server setup.

FYI @hodyroff

**Comment by mmattel** [comment]
We already have the following in our NFS documentation, pls tell if or why this is not sufficient.

https://doc.owncloud.com/server/10.7/admin_manual/installation/deployment_recommendations/nfs.html#syncasync

-- --
**sync/async**
With the default value of async, the NFS client may delay sending application writes to the NFS server. In other words, under normal circumstances, data written by an application may not immediately appear on the server that hosts the file. sync provides greater data cache coherence among clients, but at a significant performance cost. Having the database like MySQL or Mariadb on NFS, the default database option value for innodb_flush_method is fsync, even if it is not explicitly set. This database option forces the mount to immediately write to the NFS server without generally setting the mount sync option and avoiding this performance penalty. You may consider further tuning when using clustederd server environments.
-- --

Sidenote, we also have a note regarding _Start a Service After a Resource is Mounted_ which is true for NFS as well to avoid that a database or webserver cant connect because an NFS mount has not finished.

https://doc.owncloud.com/server/10.7/admin_manual/installation/manual_installation/server_prep_ubuntu_20.04.html#start-a-service-after-a-resource-is-mounted

**Comment by dragotin** [comment]
I would appreciate if the default of `async` for the NFS mount would come out more clearly. `sync` is almost never the right choice because it has a huge performance drop.

I have no inside on the database topic though but I do not think it is a good idea to have the database on NFS, is it?

**Comment by mmattel** [comment]
> I would appreciate if the default of async for the NFS mount would come out more clearly. sync is almost never the right choice because it has a huge performance drop.

The man page of mount states the following:
```
defaults
              Use the default options: rw, suid, dev, exec, auto, nouser, and async.
```
Means, that if not otherwise declared in the mount options, these are the default = async is not needed to be explcitly declared. But if you want me to do, I can drop a note about the default options.

> I have no inside on the database topic though but I do not think it is a good idea to have the database on NFS, is it?

That is a very very old opinion (+20y ago) and  a lot of fud has been spreaded around by storage vendors to protect their FiberChannel market. Except if you have a very high performance DB, you can run databases on NFS very performant and securely. NFS has a lot of benefits for databases. Of course, which is true for all environments, you need to provide a proper bandwidth and memory, but this is nothing special to NFS. We can disucss this offline if you want as I was one of the evegalists and front battle fighter at the time for running databases on NFS 😄

**Comment by mmattel** [comment]
Closed via #3939
